pandas/core/frame.py

```diff
@@ -4485,16 +4485,17 @@ def join(self, other, on=None, how='left', lsuffix='', rsuffix='',
             * left: use calling frame's index (or column if on is specified)
             * right: use other frame's index
             * outer: form union of calling frame's index (or column if on is
-                specified) with other frame's index
+                specified) with other frame's index, and sort it
```

Comment:
You are right! This also affects `DataFrame.merge`, `pd.merge` and `Series.align`. Their docstrings need to be updated. And also the API-breaking whatsnew notice!